parcels/interaction/hash_spherical.py

```diff
@@ -0,0 +1,167 @@
+from math import ceil
+
+# from numba.core.decorators import njit
+import numpy as np
+
+from parcels.interaction.base_neighbor import BaseSphericalNeighborSearch
+from parcels.interaction.base_hash import BaseHashNeighborSearch, hash_split
+
+
+class HashSphericalNeighborSearch(BaseHashNeighborSearch,
+                                  BaseSphericalNeighborSearch):
+    '''Neighbor search using a hashtable (similar to octtrees).'''
+    name = "hash"
+
+    def __init__(self, interaction_distance, interaction_depth,
+                 max_depth=100000):
+        '''Initialize the neighbor data structure.
+
+        :param interaction_distance: maximum horizontal interaction distance.
+        :param interaction_depth: maximum depth of interaction.
+        :param values: lat, long, depth values for particles.
+        :param max_depth: maximum depth of the ocean.
+        '''
+        super().__init__(interaction_distance, interaction_depth, max_depth)
+
+        self._init_structure()
+
+    def _find_neighbors(self, hash_id, coor):
+        '''Get neighbors from hash_id and location.'''
+        # Get the neighboring cells.
+        neighbor_blocks = geo_hash_to_neighbors(
+            hash_id, coor, self._bits, self.inter_arc_dist)
+        all_neighbor_points = []
+
+        # Get the particles from the neighboring cells.
+        for block in neighbor_blocks:
+            try:
+                all_neighbor_points.extend(self._hashtable[block])
+            except KeyError:
+                pass
+
+        potential_neighbors = np.array(all_neighbor_points, dtype=int)
+        return self._get_close_neighbor_dist(coor, potential_neighbors)
+
+    def _values_to_hashes(self, values, active_idx=None):
+        '''Convert coordinates to cell ids.
+
+        :param values: positions of particles to convert.
```

Comment:
in format `[lat, lon, depth]` or in format `[lon, lat, depth]` ?

Reply:
Everything is in the same [lat, long, depth].